geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegionClear.java

```diff
@@ -30,18 +31,22 @@
 import org.apache.geode.internal.cache.versions.RegionVersionVector;
 import org.apache.geode.logging.internal.log4j.api.LogService;
 
-public class ClearPartitionedRegion {
+public class PartitionedRegionClear {
 
   private static final Logger logger = LogService.getLogger();
 
+  private static final String CLEAR_OPERATION = "_clearOperation";
+
+  private final int retryTime = 2 * 60 * 1000 /* partitionedRegion.getRetryTimeout() */;
```

Comment:
This timeout still seems arbitrary. Is there a system property or default timeout that we could reference here instead of hard-coding the value? Why can't we use the value of partitionedRegion.getRetryTimeout() and set it in the constructor? Having a custom timeout just for the clear operation that is different from other region operations seems inconsistent.

Reply:
Our current clear timeout is actually determined by rvv domination timeout, i.e. 5 seconds. 

If a clear took more than 5 seconds, that means a rvv domination timeout happened. Something must have been wrong.

Reply:
The "partitionedRegion.getRetryTimeout()" is an hour long. In clear operation case, it doesn't make sense to wait for so long. Instead we throw partial clear exception and letting user to handle that.

Reply:
Gester, This is the timeout used to wait for secondary buckets to become primary...Looking at the code and the test failures i have seen, its indeterministic and there is no easy way to wait and notify on this action.

Reply:
@DonalEvans Also, adding system property, it doesn't look like it will be of any help to application and I am not convinced that this needs a system property. In this case if we are unable to find Primary bucket getting elected in a given time we are throwing partial clear exception; based on which user can take further action, like retry.

Reply:
@agingade I see. So this timeout is kind of unique to this step in the clear operation and so isn't needed/tracked anywhere else.

Reply:
Thats right.